addons/rating/models/rating.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+import logging
+import uuid
+
+from openerp import api, fields, models, _
+
+_logger = logging.getLogger(__name__)
+
+
+class Rating(models.Model):
+
+    _name = "rating.rating"
+    _description = "Rating"
+    _order = 'create_date desc'
+    _rec_name = 'res_name'
+    _sql_constraints = [
+            ('rating_range', 'check(rating >= -1 and rating <= 10)', 'Rating should be between -1 to 10'),
+    ]
+
+    @api.multi
+    @api.depends('res_model', 'res_id')
+    def _compute_res_name(self):
+        for record in self:
+            # TDE REMARKS : access right problem ?
+            name = self.env[record.res_model].sudo().browse(record.res_id).name_get()
+            record.res_name = name and name[0][1] or ('%s/%s') % (record.res_model, record.res_id)
+
+    @api.model
+    def new_access_token(self):
+        return uuid.uuid4().hex
+
+    res_name = fields.Char(compute='_compute_res_name', string='Resource Name', store=True, help="The name of the rated ressource.")
+    res_model = fields.Char(string='Resource Model', required=True, help="Model name of the rated object", index=True)
+    res_id = fields.Integer(string='Resource ID', required=True, help="Identifier of the rated object", index=True)
+    rated_partner_id = fields.Many2one('res.partner', string="Rated Partner", help="Owner of the rated ressource")
+    partner_id = fields.Many2one('res.partner', string='Customer', help="Author of the rating")
+    rating = fields.Float(string="Rating", group_operator="avg", default=-1, help="Rating value")
+    feedback = fields.Text('Feedback reason', help="Reason of the rating")
+    access_token = fields.Char(string='Security Token', default=new_access_token, help="Access token to set the rating of the value")
+
+    @api.model
+    def apply_rating(self, rate, res_model=None, res_id=None, token=None):
+        """ apply a rating for given res_model/res_id or token. If the res_model is a mail.thread
+            object, a message will be post in the chatter.
+            :param float rating : the rating value to apply
+            :param list res_id : id of the rated object.
+            :param string res_model : name of model.
+            :param string token : access token
+            :return the rating record
+        """
+        domain = [('access_token', '=', token)] if token else [('res_model', '=', res_model), ('res_id', '=', res_id)]
+        rating = self.search(domain, limit=1)
+        if rating:
```

Comment:
If no rating, nothing is done, is it intended ? For example it could create a new rating if no one is found, or is it managed by the caller ?